net/mwan3/files/lib/mwan3/common.sh

```diff
@@ -63,19 +63,29 @@ mwan3_get_src_ip()
 	if [ "$family" = "ipv4" ]; then
 		addr_cmd='network_get_ipaddr'
 		default_ip="0.0.0.0"
-		sed_str='s/ *inet \([^ \/]*\).*/\1/;T; pq'
+		sed_str='s/ *inet \([^ \/]*\).*/\1/;T;p'
 		IP="$IP4"
 	elif [ "$family" = "ipv6" ]; then
 		addr_cmd='network_get_ipaddr6'
 		default_ip="::"
-		sed_str='s/ *inet6 \([^ \/]*\).* scope.*/\1/;T; pq'
+		sed_str='s/ *inet6 \([^ \/]*\).* scope.*/\1/;T;p'
 		IP="$IP6"
 	fi
 
 	$addr_cmd _src_ip "$true_iface"
 	if [ -z "$_src_ip" ]; then
-		network_get_device device $true_iface
-		_src_ip=$($IP address ls dev $device 2>/dev/null | sed -ne "$sed_str")
+		if [ "$family" = "ipv6" ]; then
+			network_get_prefix6 _src_ip "$true_iface"
+			if [ -n "$_src_ip" ]; then
+				# got a prefix, find an interface with a delegated address
+				_src_ip=$(echo "$_src_ip" | sed -e 's;:*/.*$;;')
+				_src_ip=$($IP address ls | sed -ne "s/ *inet6 \($_src_ip:[0-6a-f:]\+\).* scope.*/\1/;T;p" | head -n1)
```

Comment:
The regexe  are always cryptic! So it would be good if you could add a comment and an example, at least in the commit. That would be useful for the future if this code section is changed again. So it can be tested